src/services/iota.js

```diff
@@ -72,69 +65,67 @@ const skinnyQueryTransactions = (iotaProvider, addresses) =>
     );
   });
 
-const initializePolling = addresses => {
-  totalLength = addresses.length;
-
-  indexes.startingIdx = 0;
-  indexes.endingIdx = addresses.length - 1;
-  indexes.latestFoundBackIdx = indexes.endingIdx;
-
-  indexes.frontIdx =
-    indexes.startingIdx + Math.floor(Math.random() * BUNDLE_SIZE);
-  indexes.backIdx = indexes.endingIdx - Math.floor(Math.random() * BUNDLE_SIZE);
-};
-
 const checkUploadPercentage = addresses => {
+  let idx = store.getState().upload.indexes;
+
   let backOfFile = new Promise((resolve, reject) => {
-    skinnyQueryTransactions(IotaA, [addresses[indexes.backIdx]]).then(
+    skinnyQueryTransactions(IotaA, [addresses[idx.backIdx]]).then(
       transactions => {
         if (transactions.length > 0) {
-          indexes.latestFoundBackIdx = indexes.backIdx;
-          updateBackIndex(indexes.backIdx);
+          store.dispatch(
+            uploadActions.updateBackIndex(
+              Math.max(
+                ...[
+                  idx.backIdx - Math.floor(Math.random() * BUNDLE_SIZE),
+                  idx.startingIdx
+                ]
+              )
+            )
+          );
         }
         resolve();
       }
     );
   });
 
   let frontOfFile = new Promise((resolve, reject) => {
-    skinnyQueryTransactions(IotaA, [addresses[indexes.frontIdx]]).then(
+    skinnyQueryTransactions(IotaA, [addresses[idx.frontIdx]]).then(
       transactions => {
         if (transactions.length > 0) {
-          indexes.latestFoundFrontIdx = indexes.frontIdx;
-          updateFrontIndex(indexes.frontIdx);
+          store.dispatch(
```

Comment:
Dispatch this from an epic after the promise resolves. Files in `/services` shouldn't have any dependencies from our business logic code.